src/ucp/wireup/wireup.c

```diff
@@ -1208,29 +1179,35 @@ ucs_status_t ucp_wireup_init_lanes(ucp_ep_h ep, unsigned ep_init_flags,
         ucs_fatal("endpoint reconfiguration not supported yet");
     }
 
+    old_cfg_index = ep->cfg_index;
     ep->cfg_index = new_cfg_index;
     ep->am_lane   = key.am_lane;
 
     snprintf(str, sizeof(str), "ep %p", ep);
     ucp_wireup_print_config(worker, &ucp_ep_config(ep)->key, str,
-                            addr_indices, cm_idx, UCS_LOG_LEVEL_INFO);
+                            addr_indices, cm_idx, UCS_LOG_LEVEL_DEBUG);
 
     /* establish connections on all underlying endpoints */
     for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
         if (ucp_ep_get_cm_lane(ep) == lane) {
-            /* restore the cm lane after reconfiguration */
-            ep->uct_eps[lane] = &cm_wireup_ep->super.super;
             continue;
         }
 
-        status = ucp_wireup_connect_lane(ep, ep_init_flags, lane,
-                                         key.lanes[lane].path_index,
-                                         remote_address, addr_indices[lane]);
-        if (status != UCS_OK) {
-            return status;
+        if ((old_cfg_index == UCP_WORKER_CFG_INDEX_NULL) ||
+            (ucp_ep_config_find_lane_index(&ep->worker->ep_config
+                                           [old_cfg_index].key, lane,
+                                           reuse_lane_map) == UCP_NULL_LANE)) {
+            status = ucp_wireup_connect_lane(ep, ep_init_flags, lane,
+                                             key.lanes[lane].path_index,
+                                             remote_address, addr_indices[lane]);
+            if (status != UCS_OK) {
+                return status;
+            }
         }
 
         ucs_assert(ep->uct_eps[lane] != NULL);
+        rsc_index = ucp_ep_get_rsc_index(ep, lane);
+        ucp_worker_iface_progress_ep(ucp_worker_iface(worker, rsc_index));
```

Comment:
why needed? isn't it done inside ucp_wireup_connect_lane already?

Reply:
it was needed to fix the issue that iface of UCT EPs created on CM phase weren't' added to progress
added missing action to add UCT iface to progress when created UCT EP on CM phase